lib/apimatic-core/utilities/api_helper.rb

```diff
@@ -442,6 +442,45 @@ def self.map_response(obj, keys)
       val
     end
 
+    # Apply unboxing_function to additional properties from hash.
+    # @param [Hash] hash The hash to extract additional properties from.
+    # @param [Proc] unboxing_function The deserializer to apply to each item in the hash.
+    # @return [Hash] A hash containing the additional properties and their values.
+    def self.get_additional_properties(hash, unboxing_function)
+      additional_properties = {}
+
+      # Iterate over each key-value pair in the input hash
+      hash.each do |key, value|
+        # If the value is a complex structure (Hash or Array), we apply apply_unboxing_function.
+        additional_properties[key] = if value.is_a?(Hash) || value.is_a?(Array)
```

Comment:
Instead of depending on the value type like `value.is_a?(Hash) || value.is_a?(Array)`, these flags should be passed in the parameters from the caller to restrict the types for additional properties. there will be is_array, is_map, is_array_of_map (this will have the both is_array and is_map flags enabled).